File: Software/SoftwareIdeasModeler/Data/Templates/Code/Ruby/Default/Class.rb
<%==SIM:ForEach:Imports==%>
require '<%==SIM:Import.Name==%>'
<%==SIM:EndFor==%>

<%==SIM:ForEach:Element.DocumentationLines==%># <%==SIM:DocumentationLine==%>
<%==SIM:EndFor==%>module <%==SIM:Element.Namespace==%>
    class <%==SIM:Element.Name==%><%==SIM:If:Element.HasSuperClass==%><<%==SIM:Element.SuperClass.Name==%><%==SIM:EndIf==%>

		attr_accessor <%==SIM:ForEach:Element.AllAttributes==%>:<%==SIM:Attribute.Name==%><%==SIM:IfNot:IsLastItem==%>, <%==SIM:EndIf==%><%==SIM:EndFor==%>

<%==SIM:ForEach:Element.Operations==%>
		def <%==SIM:Operation.Name==%> (<%==SIM:ForEach:Operation.Parameters==%><%==SIM:Parameter.Name==%><%==SIM:IfNot:IsLastItem==%>, <%==SIM:EndIf==%><%==SIM:EndFor==%>)

		end
<%==SIM:EndFor==%>
    end
end

File: Software/SoftwareIdeasModeler/Data/Templates/Code/JavaScript/Default/Class.js
<%==SIM:ForEach:Element.DocumentationLines==%>//<%==SIM:DocumentationLine==%>
<%==SIM:EndFor==%>function <%==SIM:Element.Name==%>
{
<%==SIM:ForEach:Element.AllAttributes==%>
    <%==SIM:ForEach:Attribute.DocumentationLines==%>//<%==SIM:DocumentationLine==%>
    <%==SIM:EndFor==%>this.<%==SIM:Attribute.Name==%><%==SIM:If:Attribute.HasDefaultValue==%>= <%==SIM:Attribute.DefaultValue==%><%==SIM:EndIf==%>;
<%==SIM:EndFor==%>

<%==SIM:ForEach:Element.Operations==%>
	<%==SIM:ForEach:Operation.DocumentationLines==%>//<%==SIM:DocumentationLine==%>
    <%==SIM:EndFor==%>this.<%==SIM:Operation.Name==%> = function(<%==SIM:ForEach:Operation.Parameters==%> /* <%==SIM:Parameter.Type==%> */ <%==SIM:Parameter.Name==%><%==SIM:IfNot:IsLastItem==%>, <%==SIM:EndIf==%><%==SIM:EndFor==%>) 
    {
        /* Method body */
    }

<%==SIM:EndFor==%>
}

<%==SIM:If:Element.HasSuperClass==%><%==SIM:Element.Name==%>.prototype = new <%==SIM:Element.SuperClass.Name==%>()<%==SIM:EndIf==%>
